relay-spans/src/span.rs

```diff
@@ -88,79 +94,89 @@ pub fn otel_to_sentry_span(otel_span: OtelSpan) -> EventSpan {
         _ => Some(hex::encode(parent_span_id)),
     };
 
-    let segment_id = if parent_span_id.is_none() {
-        Annotated::new(SpanId(span_id.clone()))
-    } else {
-        // TODO: derive from attributes
-        Annotated::empty()
-    };
-
     let mut op = None;
-    let mut description = name;
+    let mut description = if name.is_empty() { None } else { Some(name) };
     let mut http_method = None;
     let mut http_route = None;
     let mut http_status_code = None;
     let mut grpc_status_code = None;
+    let mut platform = None;
+    let mut segment_id = None;
+    let mut profile_id = None;
     for attribute in attributes.into_iter() {
         if let Some(value) = attribute.value.and_then(|v| v.value) {
-            let key: String = if let Some(key) = OTEL_TO_SENTRY_TAGS.get(attribute.key.as_str()) {
-                key.to_string()
-            } else {
-                attribute.key
-            };
-            if key == "sentry.op" {
-                op = otel_value_to_string(value);
-            } else if key.starts_with("db") {
-                op = op.or(Some("db".to_string()));
-                if key == "db.statement" {
-                    if let Some(statement) = otel_value_to_string(value) {
-                        description = statement;
-                    }
+            match attribute.key.as_str() {
```

Comment:
Sorry for the large diff, I converted this `if` to a `match` expression.

Reply:
Large diffs that clean up code make me happy.